Shopify/tophat
# Fix missing error handling for incompatible device artifacts

### What does this change accomplish?

This change adds error handling for incompatible artifacts for devices that was lost when the installation ticketing mechanism was created.  The purpose of these checks is to provide user visibility in case incorrect information was provided to Tophat prior to install.

### How have you achieved it?

By adding a missing `ApplicationError` case in the event that the application is not compatible with a device, and creating the localized description for it.

### How can the change be tested?

1. Add a Quick Launch entry for an iOS application, but incorrectly mark it as being an Android application.
2. Try to install the application and observe that you get a meaningful error message.
